test/lib/concerns/downstream_test.rb

```diff
@@ -0,0 +1,120 @@
+require 'test_helper'
+
+class FakeSatelliteFeature
+  include ForemanMaintain::Concerns::SystemHelpers
+  include ForemanMaintain::Concerns::Downstream
+
+  def package_name
+    'satellite'
+  end
+
+  def current_version
+    @current_version ||= version('6.10.1')
+  end
+end
+
+module ForemanMaintain
+  describe Concerns::Downstream do
+    let(:system) { FakeSatelliteFeature.new }
+
+    describe '.current_minor_version' do
+      it 'returns correct minor version' do
+        assert system.current_minor_version == '6.10'
+      end
+    end
+
+    describe '.ansible_repo' do
+      it 'returns correct ansible 2.9 repo for 6.9 on el7' do
+        system.stubs(:el_major_version).returns(7)
+        assert_equal 'rhel-7-server-ansible-2.9-rpms',
+                     system.send(:ansible_repo, system.version('6.9'))
+      end
+
+      it 'returns correct ansible 2.9 repo for 7.0 on el8' do
+        system.stubs(:el_major_version).returns(8)
+        assert_equal 'ansible-2.9-for-rhel-8-x86_64-rpms',
+                     system.send(:ansible_repo, system.version('7.0'))
+      end
+    end
+
+    describe '.product_specific_repos' do
+      it 'returns correct repos for 6.10 on el7' do
+        system.stubs(:el_major_version).returns(7)
+        expected_repos = ['rhel-7-server-satellite-6.10-rpms',
+                          'rhel-7-server-satellite-maintenance-6-rpms']
+        assert_equal expected_repos,
+                     system.send(:product_specific_repos, '6.10')
+      end
+
+      it 'returns correct repos for 7.0 on el8' do
+        system.stubs(:el_major_version).returns(8)
+        expected_repos = ['satellite-7.0-for-rhel-8-x86_64-rpms',
```

Comment:
Should we also add a test for the `satellite-capsule` repository to ensure `package_name` resolves correctly in both situations?

Reply:
oh yeah, I can add this

Reply:
done